pkgs/by-name/ho/homarr/package.nix

```diff
@@ -0,0 +1,186 @@
+{
+  bash,
+  dotenv-cli,
+  coreutils,
+  lib,
+  installShellFiles,
+  fetchFromGitHub,
+  makeWrapper,
+  inter,
+  nodejs,
+  pnpm_10,
+  python3,
+  node-gyp,
+  removeReferencesTo,
+  replaceVars,
+  srcOnly,
+  stdenv,
+}:
+let
+  pnpm = pnpm_10;
+  nodeSources = srcOnly nodejs;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "homarr";
+  version = "1.31.0";
+
+  src = fetchFromGitHub {
+    owner = "homarr-labs";
+    repo = "homarr";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-0pdZHTmubcsZivHrH31MVRSEJH6rM0846qmpqwC1dGM=";
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    fetcherVersion = 1;
+    pnpmInstallFlags = [
+      "--recursive"
+    ];
+    hash = "sha256-Il5W2iz78k6DKeNvIjLKIczWJARxdV4ic9wLAYJXbqM=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    nodejs
+    pnpm.configHook
+    python3
+    node-gyp
+  ];
+
+  propagatedBuildInputs = [
+    pnpm
+    nodejs
+  ];
+
+  patches = [
+    # nextjs tries to download google fonts from the internet during buildPhase and fails in Nix sandbox.
+    # We patch the code to expect a local font from src/app/Inter.ttf that we load from Nixpkgs in preBuild phase.
+    ./0001-font-replace-remote-google-font-with-local.patch
+    ./0002-ports-allow-variable-ports-for-web-and-websockets.patch
+    ./0003-redis-allow-custom-db.patch
+    ./0004-nodejs-relax-dep-to-22.17.patch
```

Comment:
wonder if there's a better way to relax the `engine` key in package.json to `nodejs.version`. Maybe I just need to do a replace on the engine? 22.18 isn't in yet and this'll likely continue to be an issue.